Sources/SmithyCodegenCore/Model/Model+AST.swift

```diff
@@ -0,0 +1,164 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import enum Smithy.Node
+import struct Smithy.ShapeID
+
+extension Model {
```

Comment:
Here, an extension on `Model` is provided to create one from JSON AST.

See the ample inline comments below for details.